packages/protocol/src/utils/type-util.ts

```diff
@@ -59,6 +59,11 @@ export type MaybePromise<T> = T | PromiseLike<T>;
  */
 export type TypeGuard<T> = (element: any) => element is T;
 
+/** Utility function to combine two type guards */
+export function typeGuard<T, G>(one: TypeGuard<T>, other: TypeGuard<G>): TypeGuard<T & G> {
```

Comment:
We could allow for potentially merging more than 2 guards if we spreat the other argument

Reply:
Hmm... how does it work with the Generics arguments though? Because right now it combines `T` and `G` but if we are using varargs, I don't think we can properly specify the type for the return value. But my Typescript knowledge limit my thinking here ;-)

Reply:
True. I overlooked that somehow. Please ignore the initial comment 😉